src/prefect/agent/local/agent.py

```diff
@@ -209,8 +209,17 @@ def populate_env_vars(
         # 4. Values set on the agent via `--env`
         env.update(self.env_vars)
 
-        # 5. Values set on a LocalRun RunConfig (if present
+        # 5. Values set on a LocalRun RunConfig (if present)
         if run_config is not None and run_config.env is not None:
+            # logging level passed via runconfig environment variable
+            # should always take precedence
+            env.update(
+                {
+                    "PREFECT__LOGGING__LEVEL": run_config.env.pop(
+                        "PREFECT__LOGGING__LEVEL", env["PREFECT__LOGGING__LEVEL"]
```

Comment:
does `env` always have this key?

Reply:
for the local agent it's set here https://github.com/PrefectHQ/prefect/pull/4383/files/caacdb3e8693d1d1bf4e95663220e99d240a1ff2#diff-83a67d1b4d4351a86132201d4d9794810d2b1a49aa051794d682e39daf9c67bdR193

Taking another look at it, it makes sense to make the logic a bit more uniform. After we clarify the comments above I'll do a little refactor here